File: A1 Sum to Target Number.py
def twoSum(nums, target):
    # Create a dictionary to store the complement of each number and its index
    complements = {}
    
    for i, num in enumerate(nums):
        complement = target - num
        
        # Check if the complement exists in the dictionary
        if complement in complements:
            # Return the indices of the two numbers
            return [complements[complement], i]
        
        # Store the current number and its index in the dictionary
        complements[num] = i
    
    # No solution found
    return []

# Read input values
n = int(input())
nums = list(map(int, input().split()))
target = int(input())

# Find the indices of the two numbers
result = twoSum(nums, target)

# Print the indices
print(result[0], result[1])

File: A1 Sum to Target Number.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> complements;
    
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        
        if (complements.count(complement)) {
            return {complements[complement], i};
        }
        
        complements[nums[i]] = i;
    }
    
    return {};
}

int main() {
    int n;
    cin >> n;
    
    vector<int> nums(n);
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }
    
    int target;
    cin >> target;
    
    vector<int> result = twoSum(nums, target);
    
    cout << result[0] << " " << result[1] << endl;
    
    return 0;
}
